.github/workflows/publish.yml

```diff
@@ -0,0 +1,87 @@
+name: publish artifacts
```

Comment:
This should all be gated by the test run. Otherwise, you could publish a version of the package that fails tests.

I've not done it before, but I believe it's possible to call the unittest workflow from within this workflow:
https://docs.github.com/en/actions/using-workflows/reusing-workflows